### PR TITLE
discord: handle invalid json in disable-breaking-updates

### LINKED ISSUE #206106: Discord's disable breaking updates script breaks if settings.json is invalid
### Describe the bug
Discord's package has a script which disables breaking updates, by looking and modifying the settings.json file.
This script breaks if said `settings.json` exists but is empty (fails to load the JSON object)
For some reason all values in this file cleared to me and I encountered this issue.
(tested on NixOS 22.11 but the script hasn't changed in master)
### Steps To Reproduce
1. Exit discord
2. Truncate `~/.config/discord/settings.json`
3. Launch discord
4. Discord should not launch at all (silently, unless launched from a terminal)

### Expected behavior
`disable-breaking-updates.py` should handle the exception `json.decoder.JSONDecodeError` which arises from this issue.

### Screenshots
N/A

### Additional context
Some logs of when this happens:
```txt
Traceback (most recent call last):
  File "/nix/store/r1r4a5yjkmkjwl0lp0hn9s9fg6sz9c1j-disable-breaking-updates.py/bin/disable-breaking-updates.py", line 27, in <module>
    settings = json.load(settings_file)
  File "/nix/store/zdba9frlxj2ba8ca095win3nphsiiqhb-python3-3.10.8/lib/python3.10/json/__init__.py", line 293, in load
    return loads(fp.read(),
  File "/nix/store/zdba9frlxj2ba8ca095win3nphsiiqhb-python3-3.10.8/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/nix/store/zdba9frlxj2ba8ca095win3nphsiiqhb-python3-3.10.8/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/nix/store/zdba9frlxj2ba8ca095win3nphsiiqhb-python3-3.10.8/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

### Notify maintainers

@devins2518 @Artturin @Infinidoge 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.80, NixOS, 22.11 (Raccoon), 22.11.20221130.596a8e8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `"nixos-22.11"`
 - channels(rg): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

~ took 4s 
❯ 
```

### PR DESCRIPTION
**Pull request by Artturin**
###### Description of changes
Closes https://github.com/NixOS/nixpkgs/issues/206106

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/discord/disable-breaking-updates.py [hunk 1]
```diff
@@ -25,7 +26,11 @@
 
 if os.path.exists(settings_path):
     with settings_path.open(encoding="utf-8") as settings_file:
-        settings = json.load(settings_file)
+        try:
+            settings = json.load(settings_file)
+        except json.JSONDecodeError:
+            print("[Nix] settings.json is malformed, letting Discord fix itself")
```
Comment (bhipple) [inline pkgs/applications/networking/instant-messengers/discord/disable-breaking-updates.py:32]:
```suggestion
            print(f"[Nix] {settings_file} is malformed; remove it to let Discord fix itself")
```
Perhaps tell the user where the file is and what to do?

Comment (Artturin) [reply pkgs/applications/networking/instant-messengers/discord/disable-breaking-updates.py:32]: the user doesn't need to do anything

Comment (bhipple) [reply pkgs/applications/networking/instant-messengers/discord/disable-breaking-updates.py:32]: Oh I see, my mistake.  I assumed this was a wrapper script and that the `sys.exit(0)` meant it wouldn't proceed to launch discord itself.

**Review by bhipple — APPROVED** [review]

**Comment by rafaelsgirao** [comment]
I know I'm late to the party since this has already been merged, but when/if Discord gets a breaking update and the settings file is invalid for whatever reason, won't this get the user stuck in Discord's breaking update screen? I don't think it's good to let the user get stuck there without any additional info: it sort of defeats the purpose of the script in the first place.

**Pull request merged**